src/category_theory/triangulated/pretriangulated.lean

```diff
@@ -0,0 +1,219 @@
+end
+
+/--
+Given any distinguished triangle
+```
+      f       g       h
+  X  ───> Y  ───> Z  ───> X⟦1⟧
+```
+the composition `h ≫ f⟦1⟧ = 0`.
+See https://stacks.math.columbia.edu/tag/0146
+-/
+lemma comp_dist_triangle_mor_zero₃₁  [CT : pretriangulated C] (T ∈ CT.distinguished_triangles) :
+  T.mor₃ ≫ ((shift C).functor.map T.mor₁) = 0 :=
+begin
+    have H₂ := rot_of_dist_triangle C (triangle.rotate C T) (rot_of_dist_triangle C T H),
+    have t := comp_dist_triangle_mor_zero₁₂ C (triangle.rotate C (triangle.rotate C T)) H₂,
+    dsimp at t,
+    rw [comp_neg, neg_eq_zero] at t,
+    exact t,
+end
+
+/-
+TODO: If `C` is pretriangulated with respect to a shift,
+then `C^{op}` is pretriangulated with respect to the inverse shift.
+-/
+
+end category_theory.triangulated
+
+namespace category_theory.triangulated
+
+variables (C : Type u₁) [category.{v₁} C] [has_shift C] [additive_category C]
+[functor.additive (shift C).functor] [functor.additive (shift C).inverse]
+variables (D : Type u₂) [category.{v₂} D] [has_shift D] [additive_category D]
+[functor.additive (shift D).functor] [functor.additive (shift D).inverse]
+
+/--
+A triangulated functor between pretriangulated categories C and D is a functor `F : C ⥤ D`
+together with given functorial isomorphisms `ξ X : F(X⟦1⟧) ⟶ F(X)⟦1⟧` with extra conditions
+involving images of triangles.
+-/
+structure triangulated_functor_struct extends (C ⥤ D) :=
+(natural_isom : (shift C).functor ⋙ to_functor ≅ to_functor ⋙ (shift D).functor)
+
+instance : inhabited (triangulated_functor_struct C C) :=
+⟨{ obj := λ X, X,
+  map := λ _ _ f, f,
+  natural_isom := by refl }⟩
+
+variables {C D}
+/--
+Given a `triangulated_functor_struct` we can define a function from triangles of `C` to
+triangles of `D`.
+-/
+@[simp]
+def triangulated_functor_struct.map_triangle (F : triangulated_functor_struct C D)
+  (T : triangle C) : triangle D :=
+{ obj₁ := F.obj T.obj₁,
+  obj₂ := F.obj T.obj₂,
+  obj₃ := F.obj T.obj₃,
+  mor₁ := F.map T.mor₁,
+  mor₂ := F.map T.mor₂,
+  mor₃ := F.map T.mor₃ ≫ F.natural_isom.hom.app T.obj₁ }
+
+variables (C D)
+/--
+A triangulated functor between pretriangulated categories C and D is a functor `F : C ⥤ D`
+together with given functorial isomorphisms `ξ X : F(X⟦1⟧) ⟶ F(X)⟦1⟧` such that for every
+distinguished triangle `(X,Y,Z,f,g,h)` of `C`, the triangle
+`(F(X), F(Y), F(Z), F(f), F(g), F(h) ≫ (ξ X))` is a distinguished triangle of `D`.
+See https://stacks.math.columbia.edu/tag/014V
+-/
+structure triangulated_functor [CT : pretriangulated C] [DT : pretriangulated D] :=
+(F : triangulated_functor_struct C D)
+(map_distinguished : Π (T: triangle C), (T ∈ CT.distinguished_triangles) →
+  (F.map_triangle T ∈ DT.distinguished_triangles) )
+
+instance [CT : pretriangulated C] : inhabited (triangulated_functor C C) :=
+⟨{ F:= {obj := λ X, X,
+  map := λ _ _ f, f,
+  natural_isom := by refl },
+  map_distinguished := begin
+    intros T Tdt,
+    cases T,
```

Comment:
`intros, cases` can usually be shortened using `rintros`

Reply:
I couldn't get this to work using `rintros`, but if you find a way to get it working, then I'm happy to change it :)

Reply:
I believe that `rintros ⟨_, _, _, _⟩ Tdt` works